src/components/TransactionalInfo.tsx

```diff
@@ -16,35 +16,185 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { calAmountDelta } from '@alephium/sdk'
+import { Output, Transaction } from '@alephium/sdk/api/explorer'
+import { useTranslation } from 'react-i18next'
+import { useParams } from 'react-router-dom'
 import styled, { useTheme } from 'styled-components'
 
+import { AddressHash, SimpleTx, useAddressesContext } from '../contexts/addresses'
+import { isExplorerTransaction, isSimpleTx, TransactionDirection } from '../utils/transactions'
+import AddressBadge from './AddressBadge'
 import Amount from './Amount'
+import Badge from './Badge'
+import DirectionalArrow from './DirectionalArrow'
+import IOList from './IOList'
+import TimeSince from './TimeSince'
+import Token from './Token'
 
 interface TransactionalInfoProps {
-  type: 'out' | 'in' | 'pending'
-  content: string | bigint
+  transaction: Transaction | SimpleTx
+  addressHash?: string
   className?: string
-  amount?: boolean
-  prefix?: string
+  hideLabel?: boolean
 }
 
-const TransactionalInfo = ({ content, className, amount, prefix, type }: TransactionalInfoProps) => {
+const TransactionalInfo = ({ transaction: tx, addressHash, className, hideLabel }: TransactionalInfoProps) => {
+  const { addressHash: addressHashParam = '' } = useParams<{ addressHash: AddressHash }>()
+  const _addressHash = addressHash ?? addressHashParam
+
+  const { getAddress } = useAddressesContext()
   const theme = useTheme()
-  const color = type === 'out' || type === 'pending' ? theme.font.secondary : theme.global.valid
+  const { t } = useTranslation('App')
+
+  const address = getAddress(_addressHash)
+
+  let amount: bigint | undefined = BigInt(0)
+  let timestamp = 0
+  let type: TransactionDirection
+  let outputs: Output[] = []
+
+  const token = 'alph'
+
+  if (isExplorerTransaction(tx)) {
+    amount = calAmountDelta(tx, _addressHash)
+    const amountIsBigInt = typeof amount === 'bigint'
+    type = amount && amountIsBigInt && amount < 0 ? 'out' : 'in'
+    amount = amount && (type === 'out' ? amount * BigInt(-1) : amount)
+    timestamp = tx.timestamp
+    outputs = tx.outputs || []
+  } else if (isSimpleTx(tx)) {
+    type = tx.type === 'transfer' ? 'out' : 'in'
+    amount = tx.amount
+    timestamp = tx.timestamp
+    outputs = [{ hint: 0, key: '', amount: '', address: tx.toAddress }]
+  } else {
+    throw new Error('Could not determine transaction type, all transactions should have a type')
+  }
 
   return (
-    <div className={className} color={color}>
-      {prefix && <span>{prefix}</span>}
-      {amount && content ? <Amount value={BigInt(content)} fadeDecimals color={color} /> : content}
-    </div>
+    <Container className={className} theme={theme}>
```

Comment:
To avoid passing `theme`, this component could be replaced by a `div`. Then there is no need for the `Container` component. Instead, you can replace it with `export default styled(TransactionalInfo)`... and this styled component will already have access to the props and theme.

Edit: Every styled component has access to theme actually

Reply:
I'm not a fan of the suggestion to remove the locality of `Container` since it's used by the component for its specific properties, but it's done :sparkles: 

I'm not sure why theme was being passed in the first place, it was never used :shrug: